aws-organizations-account/src/main/java/software/amazon/organizations/account/BaseHandlerStd.java

```diff
@@ -183,25 +185,33 @@ public final ProgressEvent<ResourceModel, CallbackContext> handleRetriableExcept
-                if (currentAttempt < MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION) {
+        String accountInfo = model.getAccountId() == null ? handlerRequest.getLogicalResourceIdentifier() : model.getAccountId();
+
+
+        if (actionName != AccountConstants.Action.CREATE_ACCOUNT) {
+            int currentAttempt = context.getCurrentRetryAttempt(actionName, handlerName);
+            if (currentAttempt < MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION) {
+                context.setCurrentRetryAttempt(actionName, handlerName);
+                if (actionName == AccountConstants.Action.LIST_PARENTS
+                        || actionName == AccountConstants.Action.LIST_TAGS_FOR_RESOURCE
+                        || actionName == AccountConstants.Action.DESCRIBE_ACCOUNT
+                        || actionName == AccountConstants.Action.LIST_ACCOUNTS) {
+                    logger.log(String.format("Got %s when calling %s for "
+                                    + "account [%s]. Retrying %s of %s ",
+                            e.getClass().getName(), organizationsRequest.getClass().getName(), accountInfo, currentAttempt + 1, MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION));
+                    throw RetryableException.builder().cause(e).build();
+                } else {
                     int callbackDelaySeconds = computeDelayBeforeNextRetry(currentAttempt, BASE_DELAY, RANDOMIZATION_FACTOR); // in seconds
-                    context.setCurrentRetryAttempt(actionName, handlerName);
                     logger.log(String.format("Got %s when calling %s for "
-                                                 + "account [%s]. Retrying %s of %s with callback delay %s seconds.",
-                        e.getClass().getName(), organizationsRequest.getClass().getName(), accountInfo, currentAttempt + 1, MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION, callbackDelaySeconds));
+                                    + "account [%s]. Retrying %s of %s with callback delay %s seconds.",
+                            e.getClass().getName(), organizationsRequest.getClass().getName(), accountInfo, currentAttempt + 1, MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION, callbackDelaySeconds));
                     return ProgressEvent.defaultInProgressHandler(context, callbackDelaySeconds, model);
-                } else {
-                    logger.log(String.format("All retry attempts exhausted for account [%s], return exception to CloudFormation for further handling.", accountInfo));
-                    return handleError(organizationsRequest, handlerRequest, e, proxyClient, model, context, logger);
                 }
             }
-            return handleError(organizationsRequest, handlerRequest, e, proxyClient, model, context, logger);
-        } catch (Exception exception) {
-            return ProgressEvent.failed(model, context, HandlerErrorCode.GeneralServiceException, exception.getMessage());
```

Comment:
i noticed that you removed the catch and fail for generalServiceException, why are we removing this? same for the other resource types.

Reply:
For the retry to work we have to throw the RetryableException and this catch block will consume the RetryableException and we will not see the expected behavior, also we are not expecting any checked exception the error handling methods.